packages/__docs__/buildScripts/ai-accessible-documentation/generate-ai-accessible-markdowns.mts

```diff
@@ -0,0 +1,283 @@
+  defaultValue?: {
+    value?: string
+  }
+  description?: string
+}
+
+interface ComponentData {
+  displayName: string
+  description: string
+  props: Record<string, PropDefinition>
+  category?: string
+  parent?: string
+  id?: string
+  packageName?: string
+  esPath?: string
+  title?: string
+}
+
+interface GuideData {
+  title: string
+  description: string
+  category?: string
+  order?: number
+  id?: string
+  [key: string]: any
+}
+
+type DocumentationData = ComponentData | GuideData
+
+function isComponent(data: DocumentationData): data is ComponentData {
+  return (
+    (data as ComponentData).displayName !== undefined &&
+    (data.category === 'components' ||
+      data.category === 'components/AI Components' ||
+      data.category === 'components/utilities')
+  )
+}
+
+function isGuide(data: DocumentationData): data is GuideData {
+  return (
+    (data as GuideData).title !== undefined &&
+    (data.category === 'Guides' ||
+      data.category === 'Getting Started' ||
+      data.category === 'Patterns')
+  )
+}
+
+function formatPropsMarkdownTable(text: string): string {
+  return text
+    .replace(/\\/g, '\\\\')
+    .replace(/\|/g, '\\|')
+    .replace(/\n/g, ' ')
+    .replace(/\s+/g, ' ')
+    .trim()
+}
+
+function formatPropsTypeDefinitionForMarkdownTable(type: string): string {
+  return `\`${type
+    .replace(/\\/g, '\\\\')
+    .replace(/\|/g, '\\|')
+    .replace(/\n/g, ' ')
+    .replace(/\s+/g, ' ')
+    .trim()}\``
+}
+
+function formatPropsDefaultValueForMarkdownTable(value: string): string {
+  return value ? `\`${value}\`` : '-'
+}
+
+function generatePropsTable(
+  component: ComponentData,
+  childComponents: ComponentData[] = []
+): string {
+  const { displayName, props } = component
+
+  const allProps: Array<{
+    name: string
+    data: PropDefinition
+    component: string
+  }> = []
+
+  // Collect props from parent component e.g. for Tabs
+  for (const [propName, propData] of Object.entries(props)) {
+    if (propName === 'dir') continue
+    allProps.push({
+      name: propName,
+      data: propData,
+      component: displayName
+    })
+  }
+
+  // Collect props from child components e.g. for Tabs.Panel
+  for (const child of childComponents) {
+    const childName = child.displayName.split('.').pop() || ''
+    for (const [propName, propData] of Object.entries(child.props)) {
+      if (propName === 'dir') continue
+      allProps.push({
+        name: propName,
+        data: propData,
+        component: `${displayName}.${childName}`
+      })
+    }
+  }
+
+  if (allProps.length === 0) {
+    return ''
+  }
+
+  let propsContent = `### Props\n\n`
+  propsContent += `| Component | Prop | Type | Required | Default | Description |\n`
+  propsContent += `|-----------|------|------|----------|---------|-------------|\n`
+
+  for (const prop of allProps) {
+    const type = formatPropsTypeDefinitionForMarkdownTable(
+      prop.data.tsType?.raw || prop.data.type?.name || ''
```

Comment:
I think this here is a type, it should be `prop.data.tsType?.name`. I could not find the `type` field in the JSONs. Check e.g. Avatar's `src` field